stistools/defringe/normspflat.py

```diff
@@ -41,58 +52,309 @@ def normspflat(inflat, outflat, do_cal=True, biasfile=None, darkfile=None, pixel
+                                   '{}_{}.fits'.format(rootname, output_suffix))
+            # Remove files from previous iterations of this script:
+            for old_file in [outname, trailer]:
+                if os.access(old_file, os.F_OK):
+                    os.remove(old_file)
+            res = calstis(os.path.basename(inflat), wavecal=wavecal, outroot=outroot,
+                          trailer=trailer)
+            if res != 0:
+                raise Exception('CalSTIS returned non-zero code:  {}\n' \
+                                'See log for more details:  {}'.format(res, trailer))
+            print ('File written:  {}'.format(outname))
+        finally:
+            os.chdir(old_cwd)
+    else:  # not do_cal
+        outname = inflat
+
+        # Check if user-supplied inflat has all the recommended calibrations performed:
+        keyword_warnings = []
+        for keyword in perform:
+            if hdr0[keyword] != 'COMPLETE':
+                keyword_warnings.append(keyword)
+        if keyword_warnings:
+            warnings.warn('These calibration steps should be COMPLETE:\n{}'.format(
+                ', '.join(keyword_warnings)))
+
+    # Read in the calibrated flat data:
+    data = fits.getdata(outname, ext=1)
+    numrows, numcols = np.shape(data)
 
     # Do a line-by-line cubic spline fit to the fringe flat to remove the lamp function...
 
-    # If short-slit aperture (does not start with "52X"):
-    #    Find the row with max counts in the short-slit fringe flat
-    #    Uses central 60% of columns
-    #    Does some flux-filtering
+    with fits.open(outname) as hdulist:
+
+        aperture = hdulist[0].header['APERTURE']
+        bincols = hdulist[0].header['BINAXIS1']
+        binrows = hdulist[0].header['BINAXIS2']
+        cenwave = hdulist[0].header['CENWAVE']
+
+        # If short-slit aperture (does not start with "52X"):
+        if aperture[0:3] != "52X":
+            flatdata = hdulist[1].data
+            #    Find the row with max counts in the short-slit fringe flat
+            #    Search between the middle 10% of rows and the middle 60% of columns
+            t_row = int(0.45 * numrows)
+            b_row = int(0.55 * numrows)
+            #     Uses central 60% of column
+            l_col = int(0.2 * numcols)
+            r_col = int(0.8 * numcols)
+
+            row_avgs = np.array([np.average(row) for row in flatdata[t_row:b_row, l_col:r_col]])
+            max_row_idx = np.where(row_avgs == np.max(abs(row_avgs)))[0][0] + t_row  # CL does an absolute value here
+            max_row = flatdata[max_row_idx]
+            #    Does some flux-filtering -- I think this is for determining the max rows
+
+
+        # Set rows (startrow, lastrow) to be fit according to aperture name (and possibly OPT_ELEM):
+        # '0.3X0.09', '0.2X0.06', '52X...'
+        if aperture == "0.3X0.09":
+            startrow = max_row_idx - int(4./binrows+0.25)
```

Comment:
The calls to `int` in this section should be replaced by calls to `round`, because these are `nint` in the cl script.